amaranth/sim/pysim.py

```diff
@@ -262,16 +362,28 @@ def remove_trigger(self, process, signal):
         assert process in self.slots[index].waiters
         del self.slots[index].waiters[process]
 
+    def add_memory_trigger(self, process, token):
+        index = self.memories[token]
+        self.mem_slots[index].waiters.add(process)
+
+    def remove_memory_trigger(self, process, token):
+        index = self.memories[token]
+        self.mem_slots[index].waiters.remove(process)
+
     def wait_interval(self, process, interval):
         self.timeline.delay(interval, process)
 
     def commit(self, changed=None):
         converged = True
         for signal_state in self.pending:
+            if changed is not None:
+                if isinstance(signal_state, _PyMemoryState):
```

Comment:
I feel like if we're mixing them here, we might as well mix them in `slots`.

Reply:
I'm ... not really happy with the mixing, it just sort of happened as I was writing the code.

Slots actually make sense though, I'm going to do it.

Reply:
Also `if isinstance(signal_state, _PyMemoryState):` is unholy.